### DISCUSSION TITLE
Menu Paradigm

### DISCUSSION THREAD

**Discussion by kevincoleman** [online-go/online-go.com#2371] [Ideas]
It’s probably good to have a guiding paradigm that helps us organize but also helps users make sense of the items in the menu. One such paradigm is one of actions, using verbs as the main menu categories (watch, play, chat, study &c.). Another is one of content categories, using nouns as the main menu categories (games, competitions, community, resources). Currently we’re doing both, and while it’s pretty intuitive to look at initially it makes it a little confusing where certain content lives in the sub-menus. I’m not saying we should just tear this up overnight—it would be catastrophic—but in order to set us up for success with the menu long term it might be nice to think of iterating toward something in one paradigm. Here are some examples where we use one paradigm or the other.

## Current Menu
Home (noun)

Play (verb)
- Play
- Tournaments
- Ladders

Learn (verb)
- Learn to Play Go
- Sign up for AI Game Reviews
- Puzzles
- Other Go Resources

Watch (verb)
- Games

Community (noun)
- Forums
- Chat
- Groups
- Support OGS
- About
- GitHub
- Documentation & FAQ

Tools (noun)
- Joseki
- Demo Board
- SGF Calculator
- Rating Calculator

## Action-oriented
New Game (if it’s a button they’ll know it’s an action even without the verb)

Play (current Home content)

Compete
- Tournaments
- Ladders
- Watch Games

Study
- Learn to Play Go
- Watch Games
- AI Game Reviews
- Puzzles
- Joseki
- Demo Board
- SGF Calculator
- Rating Calculator
- Other Go Resources

Interact (need a better name?)
- Forums
- Chat
- Groups
- Watch Games

Learn About Us (ugh bad title)
- Become a Supporter
- About
- GitHub
- Documentation & FAQ

Or, in a noun-based paradigm:
## Subject/category

New Game (Play Content)

My Games (Home content)

Competition
- Tournaments
- Ladders
- Watch Games

Go Tools
- Learn to Play Go
- AI Game Reviews
- Puzzles
- Joseki
- Demo Board
- SGF Calculator
- Rating Calculator
- Other Resources

Community (noun)
- Forums
- Chat
- Groups
- Watch Games

About
- Support OGS
- About OGS
- GitHub
- Documentation & FAQ

**Comment by anoek** [comment]
Food for thought for future pages, it's been on the todo list for awhile now to have a page where we show currently active streams, as well as an indicator of some sort in the nav bar that there are one or more live streams to let folks know when there are streams to tune into. Another potential page would be curated links to tutorial videos. The idea was for both of those to be under the "Watch" menu as it exists today.

**Reply by kevincoleman** [reply]
Ah that makes it make more sense. Obviously in this case I rolled it into something else because it seemed redundant as a drop-down, and maybe not worth a whole section. I look forward to developing out that watch ecosystem. Seems it would get a lot of good community involvement.

**Comment by kevincoleman** [comment]
Just checking back on this to see if you have any more thoughts on if there might be something actionable here. Happy to re-include the “Watch” menu—which might be named something like “Streams” if we went with a noun-based model.

**Comment by anoek** [comment]
With your button experiment in the works it seems like we're pretty much at the "noun" model, with the exception of what to do about Watch > Games. Ultimately though I personally don't feel there's a strong driving reason to be strict with noun vs verb - if it's clear it's clear?

> I’m not saying we should just tear this up overnight—it would be catastrophic

I second this. If we have a good reason to churn, then ok, otherwise let's not impose that on people.

**Reply by kevincoleman** [reply]
I can see the wisdom in not tearing it up. I agree that clarity is more important (frankly, having to come up with some of those names seemed difficult). Perhaps maybe we just pick a route and iterate _toward_ a model? Or we could consider small reorganizations of menus to align with a model? The first of these would be to accommodate the new button idea, so maybe that’s a small change we can test the idea with.

**Reply by anoek** [reply]
Yep I think if we test out the Play button, the Compete / Competition change makes sense along with that since the button takes away the "Play" thing.

**Comment by kevincoleman** [comment]
I’m not as familiar with our modal code, but I’m sure I could find my way through the menu itself. Is this something you would want to try to find time to help code? I’m happy to take a first stab at it.

**Reply by anoek** [reply]
Awesome go for it :) Right now my focus is on anti-cheating and anti-griefing code, so I'm going to stay focused on that.

**Reply by kevincoleman** [reply]
Right on. I’ll give it a shot. I appreciate your time on anti-cheating!
